models/payment.py

```diff
@@ -310,6 +312,23 @@ def recommended_destination(self):
             except (MultipleResultsFound, NoResultFound):
                 continue
 
+    @property
+    def customer_reference(self):
+        if self.id is None:
+            raise Exception(
+                "Customer references can only be generated for payments that have "
+                "been persisted to the database."
+            )
+
+        # Derive an ISO-11649 payment reference for EUR-currency payments
+        if self.currency == "EUR":
+            order_check_digits = mod_97_10.calc_check_digits(f"{self.bankref}RF")
+            customer_reference = f"RF{order_check_digits}{self.bankref}"
+            assert iso11649.validate(customer_reference)
```

Comment:
This should be `iso11649.is_valid` I think

Reply:
Perhaps it would also make sense to move that onto a `@validates` method on the SQLAlchemy model?

Reply:
Nope, never mind: the structured reference isn't persisted (...yet?).  This could potentially be added in future.